tests/regression_tests/statepoint_restart/test.py

```diff
@@ -61,24 +61,37 @@ def test_statepoint_restart():
     harness.main()
 
 
-def test_batch_check(request):
+def test_batch_check(request, capsys):
     xmls = list(request.path.parent.glob('*.xml'))
 
     with cdtemp(xmls):
         model = openmc.Model.from_xml()
         model.settings.particles = 100
+
         # run the model
-        sp_file = model.run()
+        sp_file = model.run(export_model_xml=False)
+        assert sp_file is not None
 
         # run a restart with the resulting statepoint
         # and the settings unchanged
-        with pytest.raises(RuntimeError, match='is smaller than the number of batches'):
-            model.run(restart_file=sp_file)
-
-        # update the number of batches and run again
+        model.settings.batches = 6
+        # ensure we capture output only from the next run
+        capsys.readouterr()
+        sp_file = model.run(export_model_xml=False, restart_file=sp_file)
+        # indicates that a new statepoint file was not created
+        assert sp_file is None
+
+        output = capsys.readouterr().out
+        output = re.sub(' +', ' ', output.replace('\n',' '))
```

Comment:
Why did you need to do this? I don't understand it, sorry. It seems like you can just look for the strings below without this line.

Reply:
Ah, that was a hold-over from a different set of strings I was using to perform the check. Removing now (along with the unecessary `print` in the line below).